### PR TITLE
Fix 'with-ssl' naming issue in configure script

### PR DESCRIPTION
**Pull request by xyan264**
Also move 'pem.h' header file checking into 'with-openssl' condition.

### CODE REVIEW

**Comment by xyan264** [comment]
Commit message is updated to describe the issue: '--with-openssl=no' did not work. The issue was caused by different names -- argument defined as 'with_openssl' while testing for 'with_ssl'. 

I also took the opportunity to move the openssl header file checking into the condition clause, and combined the two testing into one single clause. The code looks cleaner this way. 

I put these in one commit since they both work on the same `with_openssl` test clause, and the number of lines are manageable. This way we can update the `configure` file once. Otherwise we will have two commits, each modifying the `configure` script. 

I can split the commit if you prefer.

File: configure [hunk 1]
```diff
@@ -3528,7 +3528,9 @@ fi
 
 done
 
-for ac_header in openssl/pem.h
+
+if test "x$with_openssl" != xno; then :
+  for ac_header in openssl/pem.h
```
Comment (kelvin-cao) [inline configure:3533]: Intent need to be adjusted.

File: configure [hunk 2]
```diff
@@ -3540,9 +3542,7 @@ fi
 
 done
 
-
-if test "x$with_ssl" != xno; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for RSA_free in -lcrypto" >&5
+        { $as_echo "$as_me:${as_lineno-$LINENO}: checking for RSA_free in -lcrypto" >&5
```
Comment (kelvin-cao) [inline configure:3545]: Is this leading white spaces intended?

File: configure [hunk 3]
```diff
@@ -3587,10 +3587,7 @@ _ACEOF
 
 fi
 
-fi
-
-if test "x$with_ssl" != xno; then :
-  ac_fn_c_check_decl "$LINENO" "RSA_get0_key" "ac_cv_have_decl_RSA_get0_key" "#include <openssl/pem.h>
+        ac_fn_c_check_decl "$LINENO" "RSA_get0_key" "ac_cv_have_decl_RSA_get0_key" "#include <openssl/pem.h>
```
Comment (kelvin-cao) [inline configure:3590]: Same as above.

**Comment by xyan264** [comment]
I think we only need to focus on `configure.ac` file since `configure` script is auto-generated from it.

The indentation in `configure` is not the best, but maybe we can live with it?

**Comment by kelvin-cao** [comment]
> 
> 
> I think we only need to focus on `configure.ac` file since `configure` script is auto-generated from it.
> 
> The indentation in `configure` is not the best, but maybe we can live with it?

Ooh, I just realized it, sure.

**Pull request merged**